src/backend/tests/expunger/test_time_analyzer.py

```diff
@@ -238,3 +238,80 @@ def test_mrc_is_eligible_in_two_years(self):
         self.run_expunger(one_year_old_conviction, nine_year_old_conviction)
 
         assert one_year_old_conviction.expungement_result.time_eligibility.date_will_be_eligible == eligibility_date
+
+
+class TestViolations(unittest.TestCase):
```

Comment:
I think this effectively duplicates a number of tests but I don't have the time to double check right now. Since the bug is about fragile string comparisons, I think it would be enough to adjust the string of an existing test.

Reply:
You're right, there are some tests for this. I'll take another look and update